src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class TagHelperAttribute : IReadOnlyTagHelperAttribute
+    {
+        public TagHelperAttribute([NotNull] string key, object value)
+        {
+            Key = key;
+            Value = value;
+        }
+
+        public string Key
+        {
+            get;
+            [param: NotNull]
+            set;
+        }
+
+        public object Value { get; set; }
+
+        public bool Equals(IReadOnlyTagHelperAttribute other)
```

Comment:
when implementing `IEquatable<IReadOnlyTagHelperAttribute>`, also override `Equals(object)` and `GetHashCode()` to match.